fenix/app/src/test/java/org/mozilla/fenix/tabstray/TabSheetBehaviorManagerTest.kt

```diff
@@ -67,35 +68,409 @@ class TabSheetBehaviorManagerTest {
+
+        // Dialog
+        val tabsTrayDialog = mockk<TabsTrayDialog>(relaxed = true)
+
+        // Fab
+        val rootView = mockk<View>(relaxed = true)
+        every { rootView.bottom } returns 1000
+        val fab = mockk<ExtendedFloatingActionButton>(relaxed = true)
+        every { fab.rootView } returns rootView
+
+        val callback = TraySheetBehaviorCallback(behavior, mockk(), tabsTrayDialog, fab)
+
+        val bottomSheet = mockk<View>(relaxed = true)
+        every { bottomSheet.top } returns 300
+        // (1000 - 300) / 1000 = 0.7
+
+        callback.onSlide(bottomSheet, 1f)
+
+        verify(exactly = 0) { tabsTrayDialog.window?.setDimAmount(any()) }
+    }
+
+    @Test
+    fun `GIVEN converted dim value is at max dim WHEN onSlide is called THEN it does not set the dialog dim amount`() {
+        // Behavior
+        val behavior = mockk<BottomSheetBehavior<ConstraintLayout>>(relaxed = true)
+
+        // Dialog
+        val tabsTrayDialog = mockk<TabsTrayDialog>(relaxed = true)
+
+        // Fab
+        val rootView = mockk<View>(relaxed = true)
+        every { rootView.bottom } returns 1000
+        val fab = mockk<ExtendedFloatingActionButton>(relaxed = true)
+        every { fab.rootView } returns rootView
+
+        val callback = TraySheetBehaviorCallback(behavior, mockk(), tabsTrayDialog, fab)
+
+        val bottomSheet = mockk<View>(relaxed = true)
+        every { bottomSheet.top } returns 400
+        // (1000 - 400) / 1000 = 0.6
+
+        callback.onSlide(bottomSheet, 1f)
+
+        verify(exactly = 0) { tabsTrayDialog.window?.setDimAmount(any()) }
+    }
+
+    @Test
+    fun `GIVEN converted dim value is less than max dim WHEN onSlide is called THEN it sets the dialog dim amount`() {
+        // Behavior
+        val behavior = mockk<BottomSheetBehavior<ConstraintLayout>>(relaxed = true)
+
+        // Dialog
+        val tabsTrayDialog = mockk<TabsTrayDialog>(relaxed = true)
+
+        // Fab
+        val rootView = mockk<View>(relaxed = true)
+        every { rootView.bottom } returns 1000
+        val fab = mockk<ExtendedFloatingActionButton>(relaxed = true)
+        every { fab.rootView } returns rootView
+
+        val callback = TraySheetBehaviorCallback(behavior, mockk(), tabsTrayDialog, fab)
+
+        val bottomSheet = mockk<View>(relaxed = true)
+        every { bottomSheet.top } returns 500
+        // (1000 - 500) / 1000 = 0.5
+
+        callback.onSlide(bottomSheet, 1f)
+
+        verify(exactly = 1) { tabsTrayDialog.window?.setDimAmount(any()) }
+    }
+
+    @Test
+    fun `GIVEN behaviour state is 'dragging' & draggedLowestSheetTop is null WHEN onSlide is called THEN draggedLowestSheetTop is set to currentBottomSheetTop`() {
+        // Behavior
+        val behavior = mockk<BottomSheetBehavior<ConstraintLayout>>(relaxed = true)
+        every { behavior.state } returns STATE_DRAGGING
+
+        // Dialog
+        val tabsTrayDialog = mockk<TabsTrayDialog>(relaxed = true)
+
+        // Fab
```

Comment:
nit: The comments for the initialization blocks (behavior, dialog, fab) in these test aren't really necessary.